packages/jsii-pacmak/test/expected.jsii-calc-base/java/src/main/java/software/amazon/jsii/tests/calculator/base/BaseProps.java

```diff
@@ -43,48 +44,75 @@ public Builder withFoo(final software.amazon.jsii.tests.calculator.baseofbase.Ve
-                private final java.lang.String $bar = java.util.Objects.requireNonNull(_bar, "bar is required");
-                private final software.amazon.jsii.tests.calculator.baseofbase.Very $foo = java.util.Objects.requireNonNull(_foo, "foo is required");
-
-                @Override
-                public java.lang.String getBar() {
-                    return this.$bar;
-                }
-
-                @Override
-                public software.amazon.jsii.tests.calculator.baseofbase.Very getFoo() {
-                    return this.$foo;
-                }
-
-                public com.fasterxml.jackson.databind.JsonNode $jsii$toJson() {
-                    com.fasterxml.jackson.databind.ObjectMapper om = software.amazon.jsii.JsiiObjectMapper.INSTANCE;
-                    com.fasterxml.jackson.databind.node.ObjectNode obj = com.fasterxml.jackson.databind.node.JsonNodeFactory.instance.objectNode();
-                    obj.set("bar", om.valueToTree(this.getBar()));
-                    obj.set("foo", om.valueToTree(this.getFoo()));
-                    return obj;
-                }
-
-            };
+            return new Jsii$Proxy(bar, foo);
         }
+
     }
 
     /**
-     * A proxy class which represents a concrete javascript instance of this type.
+     * An implementation for {@link BaseProps}
      */
-    final static class Jsii$Proxy extends software.amazon.jsii.JsiiObject implements software.amazon.jsii.tests.calculator.base.BaseProps {
-        protected Jsii$Proxy(final software.amazon.jsii.JsiiObject.InitializationMode mode) {
-            super(mode);
+    final class Jsii$Proxy extends software.amazon.jsii.JsiiObject implements BaseProps {
+        private final java.lang.String bar;
+        private final software.amazon.jsii.tests.calculator.baseofbase.Very foo;
+
+        /**
+         * Constructor that initializes the object based on values retrieved from the JsiiObject.
+         * @param objRef Reference to the JSII managed object.
+         */
+        protected Jsii$Proxy(final software.amazon.jsii.JsiiObjectRef objRef) {
+            super(software.amazon.jsii.JsiiObject.InitializationMode.JSII);
+            this.setObjRef(objRef);
+            this.bar = this.jsiiGet("bar", java.lang.String.class);
+            this.foo = this.jsiiGet("foo", software.amazon.jsii.tests.calculator.baseofbase.Very.class);
+        }
+
+
+        /**
+         * Constructor that initializes the object based on literal property values passed by the {@link Builder}.
+         */
+        private Jsii$Proxy(java.lang.String bar, software.amazon.jsii.tests.calculator.baseofbase.Very foo) {
+            super(software.amazon.jsii.JsiiObject.InitializationMode.JSII);
+            this.bar = java.util.Objects.requireNonNull(bar, "bar is required");
+            this.foo = java.util.Objects.requireNonNull(foo, "foo is required");
         }
 
         @Override
         public java.lang.String getBar() {
```

Comment:
Need the "get/set" in the proxy classes?

Reply:
Not sure I understand the comment. We need to implement the getters that have been defined on the interface...

Reply:
We resolved this offline. No change planned.